whir/src/whir/batch/committer.rs

```diff
@@ -1,184 +1,248 @@
+    pub(crate) root: Digest<E>,
+    pub(crate) ood_points: Vec<E>,
+    pub(crate) ood_answers: Vec<E>,
 }
 
-impl<F, MerkleConfig: Config> From<Witness<F, MerkleConfig>> for Witnesses<F, MerkleConfig> {
-    fn from(witness: Witness<F, MerkleConfig>) -> Self {
-        Self {
-            polys: vec![witness.polynomial],
-            merkle_tree: witness.merkle_tree,
-            merkle_leaves: witness.merkle_leaves,
-            ood_points: witness.ood_points,
-            ood_answers: witness.ood_answers,
-        }
+impl<E: ExtensionField> Witnesses<E> {
+    pub fn merkle_tree(&self) -> &MerkleTreeExt<E> {
+        &self.merkle_tree
     }
-}
 
-impl<F: Clone, MerkleConfig: Config> From<Witnesses<F, MerkleConfig>> for Witness<F, MerkleConfig> {
-    fn from(witness: Witnesses<F, MerkleConfig>) -> Self {
-        Self {
-            polynomial: witness.polys[0].clone(),
-            merkle_tree: witness.merkle_tree,
-            merkle_leaves: witness.merkle_leaves,
-            ood_points: witness.ood_points,
-            ood_answers: witness.ood_answers,
+    pub fn root(&self) -> Digest<E> {
+        self.root.clone()
+    }
+
+    pub fn to_commitment_in_transcript(&self) -> WhirCommitmentInTranscript<E> {
+        WhirCommitmentInTranscript {
+            root: self.root(),
+            ood_points: self.ood_points.clone(),
+            ood_answers: self.ood_answers.clone(),
         }
     }
+
+    pub fn num_vars(&self) -> usize {
+        self.polys[0].num_vars()
+    }
 }
 
-impl<F, MerkleConfig, PowStrategy> Committer<F, MerkleConfig, PowStrategy>
+impl<E: ExtensionField> Committer<E>
 where
-    F: FftField,
-    MerkleConfig: Config<Leaf = [F]>,
-    PowStrategy: Sync,
+    Digest<E>: IntoIterator<Item = E::BaseField> + PartialEq,
 {
-    pub fn batch_commit<Merlin>(
+    pub fn batch_commit(
         &self,
-        merlin: &mut Merlin,
-        polys: &[CoefficientList<F::BasePrimeField>],
-    ) -> ProofResult<Witnesses<F, MerkleConfig>>
-    where
-        Merlin: FieldWriter<F> + FieldChallenges<F> + ByteWriter + DigestWriter<MerkleConfig>,
-    {
-        let timer = start_timer!(|| "Batch Commit");
-        let base_domain = self.0.starting_domain.base_domain.unwrap();
-        let expansion = base_domain.size() / polys[0].num_coeffs();
-        let expand_timer = start_timer!(|| "Batch Expand");
+        polys: witness::RowMajorMatrix<E::BaseField>,
+    ) -> Result<(Witnesses<E>, WhirCommitmentInTranscript<E>), Error> {
+        let mut transcript = BasicTranscript::<E>::new(b"commitment");
+        let polys = polys.to_mles();
+        let timer = entered_span!("Batch Commit");
+        let expansion = self.0.starting_domain.size() / polys[0].evaluations().len();
+        let expand_timer = entered_span!("Batch Expand");
         let evals = polys
             .par_iter()
-            .map(|poly| expand_from_coeff(poly.coeffs(), expansion))
+            .map(|poly| {
+                expand_from_coeff(
```

Comment:
Are we able to optimize this part via `dft.dft_batch()` follow basefold [rs](https://github.com/scroll-tech/ceno/blob/b22576c0f12b512d71c93ae3ae6afeac37a558bd/mpcs/src/basefold/encoding/rs.rs#L205-L209)